src/classes/CCON_PreventUpdate_TEST.cls

```diff
@@ -0,0 +1,207 @@
+/*
+    Copyright (c) 2019, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Course Enrollments
+* @group-content ../../ApexDocContent/CourseEnrollments.htm
+* @description Unit tests for CCON_PreventUpdate_TDTM. These tests
+* make sure Contact or Course Offering on a Course Connection do not
+* change when Term Grades or Attendance Events are assocaited it.
+*/
+
+
+@isTest
+private class CCON_PreventUpdate_TEST {
+    
+    @testSetup
+    static void dataSetup() {
+    
+        //Insert a contact
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+        
+        // Insert a Course Offering
+        Course_offering__c courseOffering = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        
+        // Insert Course Connection
+        course_Enrollment__c courseEnrollment = UTIL_UnitTestData_TEST.getCourseConnection(con.Id, courseOffering.Id);
+        insert courseEnrollment;
+        
+    }
+    
+    // Test Contact and Course Offering updates with no related records on Course Connection 
+    
+    @isTest 
+    static void testUpdateCourseConnWithNoRelatedRecords() {
+    
+        //Insert a contact
+        Contact newCon = New Contact();
```

Comment:
Maybe we can create a method in the UTIL_UnitTestData_TEST class that creates a contact but you can pass in a number in order to make the data unique?

Different than the method that creates X number of contacts since we only need 1, but it would still help fix the DUPLICATES_DETECTED issue you were working on earlier.